dotnet/src/Connectors/Connectors.Memory.DuckDB/Database.cs

```diff
@@ -176,32 +167,33 @@ private static string EncodeFloatArrayToString(float[]? data)
         using var dataReader = await cmd.ExecuteReaderAsync(cancellationToken).ConfigureAwait(false);
         if (await dataReader.ReadAsync(cancellationToken).ConfigureAwait(false))
         {
-            string metadata = dataReader.GetString(dataReader.GetOrdinal("metadata"));
-            string embeddingAsString = dataReader.GetString(dataReader.GetOrdinal("embeddingAsString"));
-            string timestamp = dataReader.GetString(dataReader.GetOrdinal("timestamp"));
+            string metadata = dataReader.GetFieldValue<string?>("metadata") ?? string.Empty;
+            float[] embeddingFromSearch = (dataReader.GetFieldValue<List<float>?>("embedding")?.ToArray()) ?? Array.Empty<float>();
+            string timestamp = dataReader.GetFieldValue<string?>("timestamp") ?? string.Empty;
```

Comment:
It won't work for `string?`, only for nullable types.

Reply:
ok